nrf_modem: release v1.4.0 (#577)
nrfconnect/sdk-nrfxlib: nrf_modem/doc/CHANGELOG.rst

All notable changes to this project are documented in this file.

nrf_modem 1.4.0
***************

* The PDN socket has been removed.
* The GNSS socket has been removed.
* nrf_errno errno values have been aligned with those of newlibc.
* The :ref:`Modem API <nrf_modem_api>` (:file:`nrf_modem.h`) has been updated to return negative errno values on error.
* The :ref:`Full Modem DFU API <nrf_modem_full_dfu_api>` (:file:`nrf_modem_full_dfu.h`) has been updated to return negative errno values on error.
* The :ref:`GNSS API <nrf_modem_gnss_api>` (:file:`nrf_modem_gnss.h`) has been updated to return negative errno values on error.
* The :c:func:`nrf_modem_gnss_init` and :c:func:`nrf_modem_gnss_deinit` functions have been removed.
* Added the GNSS velocity estimate validity bit ``NRF_MODEM_GNSS_PVT_FLAG_VELOCITY_VALID``.
* Added the GNSS delete bitmask ``NRF_MODEM_GNSS_DELETE_GPS_TOW_PRECISION`` for time-of-week precision estimate.
* Added support for several new fields in the GNSS PVT notification.

> * Added support for GNSS A-GPS data need query.

Reviewer: ```suggestion
* Added support for checking when A-GPS data is needed.
```
